spring-ai-core/src/main/java/org/springframework/ai/tool/annotation/ToolParam.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2023-2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.ai.tool.annotation;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * Marks a tool argument.
+ *
+ * @author Thomas Vitale
+ * @since 1.0.0
+ */
+@Target({ ElementType.PARAMETER, ElementType.FIELD, ElementType.ANNOTATION_TYPE })
+@Retention(RetentionPolicy.RUNTIME)
+@Documented
+public @interface ToolParam {
```

Comment:
For the naming, I followed the most used convention in Spring Framework (`@RequestParam` `@CookieParam...`).

I made it tool-specific, but this could actually be used also when using Structured Outputs (as well as the entire JSON schema generation logic in `JsonSchemaGenerator`). But if we want that, we need to come up with a better name).